### PR TITLE
Add handler for Apache Derby

### PR DESCRIPTION
**Pull request by theWolf1198**
## Description

Please include a summary of the change and the issue it solves. 

**Fixes** #4793 

## Type of change

- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

### What is the solution?

Added a handler to get data from the Apache Derby database. The handler establishes a JDBC connection with the database and extracts data using SQL commands. 

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation, or created issues to update them.
- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/derby_handler/derby_handler.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        
+        self.kwargs = kwargs
+        self.parser = parse_sql
+        self.database = connection_data['database']
+        self.connectionData = connection_data
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/derby_handler/derby_handler.py:40]: Can you please use snake_case convention for naming variables?

File: mindsdb/integrations/handlers/derby_handler/derby_handler.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    def query(self, query: ASTNode) -> StatusResponse:
+        """Render and execute a SQL query.
+
+        Args:
+            query (ASTNode): The SQL query.
+
+        Returns:
+            Response: The query result.
+        """
+        if isinstance(query, ASTNode):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/derby_handler/derby_handler.py:172]: Did you check if there is a sqlalchemy dialect for derby available?

File: mindsdb/integrations/handlers/derby_handler/derby_handler.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        Args:
+            table_name (str): Name of the table to retrieve details of.
+
+        Returns:
+            Response: Details of the table.
+        """
+
+        query = f''' SELECT COLUMNNAME FROM SYS.SYSCOLUMNS INNER JOIN SYS.SYSTABLES ON SYS.SYSCOLUMNS.REFERENCEID=SYS.SYSTABLES.TABLEID WHERE TABLENAME='{table_name}' '''
+        return self.native_query(query)
+
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/derby_handler/derby_handler.py:207]: Please use two blank lines

File: mindsdb/integrations/handlers/derby_handler/derby_handler.py [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+
+
+import pandas as pd
+import jaydebeapi as jdbcconnector
+
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/derby_handler/derby_handler.py:18]: Please use two blank lines

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/derby_handler/derby_handler.py (hunk 2 above)
Comment (theWolf1198) [reply mindsdb/integrations/handlers/derby_handler/derby_handler.py:172]: I did check this, but there is no sqlalchemy dialect for derby right now. Refer: [Link](https://github.com/sqlalchemy/sqlalchemy/issues/4839)

**Comment by theWolf1198** [comment]
@ZoranPandovski, Review comments addressed.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**